inc/Addon/Cloudflare/Subscriber.php

```diff
@@ -370,15 +372,29 @@ public function save_cloudflare_options( $old_value, $value ) {
 			delete_transient( 'rocket_cloudflare_is_api_keys_valid' );
 			$is_api_keys_valid_cloudflare = $this->cloudflare->is_api_keys_valid( $value['cloudflare_email'], $value['cloudflare_api_key'], $value['cloudflare_zone_id'], true );
 			set_transient( 'rocket_cloudflare_is_api_keys_valid', $is_api_keys_valid_cloudflare, 2 * WEEK_IN_SECONDS );
-			if ( is_wp_error( $is_api_keys_valid_cloudflare ) ) {
-				$cloudflare_error_message = $is_api_keys_valid_cloudflare->get_error_message();
-				add_settings_error( 'general', 'cloudflare_api_key_invalid', __( 'WP Rocket: ', 'rocket' ) . '</strong>' . $cloudflare_error_message . '<strong>', 'error' );
-			}
+			$submit_cloudflare_view = true;
+		}
+
+		if ( ( isset( $old_value['cloudflare_devmode'], $value['cloudflare_devmode'] ) && (int) $old_value['cloudflare_devmode'] !== (int) $value['cloudflare_devmode'] ) ||
+			( isset( $old_value['cloudflare_auto_settings'], $value['cloudflare_auto_settings'] ) && (int) $old_value['cloudflare_auto_settings'] !== (int) $value['cloudflare_auto_settings'] ) ) {
+			$submit_cloudflare_view = true;
+		}
+
+		// Revalidate Cloudflare credentials if transient is false.
+		if ( false === $is_api_keys_valid_cloudflare ) {
+			$is_api_keys_valid_cloudflare = $this->cloudflare->is_api_keys_valid( $value['cloudflare_email'], $value['cloudflare_api_key'], $value['cloudflare_zone_id'] );
+			set_transient( 'rocket_cloudflare_is_api_keys_valid', $is_api_keys_valid_cloudflare, 2 * WEEK_IN_SECONDS );
+		}
+
+		// If is submit CF view & CF Credentials are invalid, display error and bail out.
+		if ( is_wp_error( $is_api_keys_valid_cloudflare ) && $submit_cloudflare_view ) {
+			$cloudflare_error_message = $is_api_keys_valid_cloudflare->get_error_message();
+			add_settings_error( 'general', 'cloudflare_api_key_invalid', __( 'WP Rocket: ', 'rocket' ) . '</strong>' . $cloudflare_error_message . '<strong>', 'error' );
```

Comment:
About `add_settings_error()`:
- This will be triggered even when `update_option()` is used directly, not only when the settings are saved through the settings page. I guess it will leads to the message appearing at random times.
- `__( 'WP Rocket: ', 'rocket' ) . '</strong>' . $cloudflare_error_message . '<strong>'`: do we know what this message may contain? If it's clearly mentioning Cloudflare it's fine, but if it's something like "Connection error" we will have a confusing message.

Reply:
The error message will be clear directly from Cloudflare. 
The one which might be confusing is related to disabled curl.